### PR TITLE
[fx][normalize] Allow for args to be left as args

### PR DESCRIPTION
**Pull request by jfix71**
Summary: Normalization is kind of broken currently. But making default arguments visible still appears to work, and is nice functionality to still be able to rely on/use. Adds an option to `NormalizeArgs`'s `__init__` called `normalize_to_only_use_kwargs` which defaults to true, which if set to false will keep using the same signature as provided, but additionally set kwargs in kwargs.

Test Plan: Added test to `test_fx_experimental`.

Differential Revision: D27759448

### CODE REVIEW

File: torch/fx/experimental/normalize.py [hunk 1]
```diff
@@ -80,13 +83,13 @@ def call_module(self, target : Target, args : Tuple[Argument, ...], kwargs : Dic
             classname = submod.__class__.__name__
             if getattr(torch.nn, classname, None) == submod.__class__:
                 sig = inspect.signature(inspect.unwrap(submod.forward))
-                new_kwargs = self._args_kwargs_to_normalized_kwargs(sig, args, kwargs)
-                if new_kwargs:
-                    return super().call_module(target, (), new_kwargs)
+                new_args, new_kwargs = self._args_kwargs_to_normalized_args_kwargs(sig, args, kwargs)
```
Comment (ansley) [inline torch/fx/experimental/normalize.py:86]: Will tuple unpacking fail if you return `None`?

File: torch/fx/experimental/normalize.py [hunk 2]
```diff
@@ -80,13 +83,13 @@ def call_module(self, target : Target, args : Tuple[Argument, ...], kwargs : Dic
...
+                if new_args is not None and new_kwargs is not None:
+                    return super().call_module(target, new_args, new_kwargs)
         return super().call_module(target, args, kwargs)
 
-    def _args_kwargs_to_normalized_kwargs(self, sig : inspect.Signature, args : Tuple[Argument, ...],
-                                          kwargs : Dict[str, Any]) -> Optional[Dict[str, Any]]:
+    def _args_kwargs_to_normalized_args_kwargs(self, sig : inspect.Signature, args : Tuple[Argument, ...],
+                                          kwargs : Dict[str, Any]) -> Optional[Tuple[Tuple[Any], Dict[str, Any]]]:
```
Comment (jamesr66a) [inline torch/fx/experimental/normalize.py:92]: This return type is getting pretty complex. Can we define a class or NamedTuple for the return?

File: torch/fx/experimental/normalize.py [hunk 3]
```diff
@@ -27,13 +28,15 @@ class NormalizeArgs(Transformer):
         traced = NormalizeArgs(traced).transform()
     """
     def __init__(self, module : torch.nn.Module, normalize_functionals : bool = True,
-                 normalize_modules : bool = True):
+                 normalize_modules : bool = True,
+                 normalize_to_only_use_kwargs : bool = True):
         super().__init__(module)
         self.normalize_functionals = normalize_functionals
         self.normalize_modules = normalize_modules
+        self.normalize_to_only_use_kwargs = normalize_to_only_use_kwargs
```
Comment (jamesr66a) [inline torch/fx/experimental/normalize.py:36]: I think this will still be broken for operators. Can you add a flag `self.normalize_operators` that defaults to False here?

Comment (jfix71) [reply torch/fx/experimental/normalize.py:36]: I'm a little confused on this one -- AFAICT `NormalizeArgs` and `NormalizeOperators` are totally independent. What would `self.normalize_operators` do inside `NormalizeArgs`?

Comment (jfix71) [reply torch/fx/experimental/normalize.py:43]: Friendly ping @jamesr66a 🙂

Comment (jamesr66a) [reply torch/fx/experimental/normalize.py:43]: The `else` branch starting at line 65 normalizes arguments for nodes with operator targets. `NormalizeOperators` changes operator targets, not args

Comment (jfix71) [reply torch/fx/experimental/normalize.py:43]: Got it, added.

**Comment by jfix71** [comment]
Reimplemented on top of https://github.com/pytorch/pytorch/pull/55992

**Review by jamesr66a — APPROVED** [review]
Looks good, thanks!
